core/src/main/java/org/apache/iceberg/rest/responses/FetchPlanningResultResponse.java

```diff
@@ -71,13 +71,25 @@ public Map<Integer, PartitionSpec> specsById() {
     return specsById;
   }
 
+  public static Builder builder() {
+    return new Builder();
+  }
+
   @Override
   public void validate() {
     Preconditions.checkArgument(planStatus() != null, "Invalid status: null");
+    Preconditions.checkArgument(
+        planStatus() == PlanStatus.COMPLETED || (planTasks() == null && fileScanTasks() == null),
```

Comment:
Im not sure if this is the correct check, i think we can try running the response tests to confirm tho.